frontend/public/components/namespace.jsx

```diff
@@ -1044,106 +1110,104 @@ const namespaceBarDropdownDispatchToProps = (dispatch) => ({
   showStartGuide: (show) => dispatch(setFlag(FLAGS.SHOW_OPENSHIFT_START_GUIDE, show)),
 });
 
-class NamespaceBarDropdowns_ extends React.Component {
-  componentDidUpdate() {
-    const { namespace, showStartGuide } = this.props;
+const NamespaceBarDropdowns_ = (props) => {
```

Comment:
@spadgett should we take this opportunity to move NamespaceBar and related components to TypeScript and address some tech debt? We can use react-redux hooks and some of our other custom hooks to greatly simplify this implementation and eliminate Firehose and HOCs used here. This file is over 1300 LOC which is a bit big.

Reply:
How big of a change is it? @kdoberst opinion? Maybe move this component into its own file in console-app and convert that to TypeScript?

Reply:
It depends on how much we address. Here's some of the big ones that I see:
- Move NamespaceBar and all related components to a new tsx file and add type definitions
- Remove all usage of redux connect HOC pattern and migrate to either custom hooks or redux `useSelector` and `useDispatch` hooks. We currently use redux connect to provide the following props to NamespaceBar/NamespaceBarDropdowns:
  - useProjects 
  - canListNS
  - canCreateProject
  - showStartGuide
- Remove Firehose and fetch namespaces/projects inside the new NamespaceDropdown component
- Move other namespace/project dropdown related state down into the new NamespaceDropdown component.

Reply:
It is pretty clear that the entire namespace.jsx needs a good refactor.  I'm thinking that this is should be a separate effort to address the entire file.  The pieces for the drop down that are new .tsx files (namespace-dropdown.tsx and namespace-menu-toggle.tsx) can easily be moved to console-app now.

I can timebox moving the rest of the namespace bar items in namespace.jsx into separate files.  If it looks like it is going to take too long or run into to many "gotcha's" we'll wait for the refactor of the rest of namespace.jsx

Reply:
Yeah, if we can get NamespaceBar and NamespaceBarDropdowns out of this file, remove Firehose and redux connect I'd be happy with that.